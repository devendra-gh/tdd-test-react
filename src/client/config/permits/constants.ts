import { IVariables } from '@tamm/app-composer';
import { PATH_ACCOUNT_UPGRADE, PATH_LOGIN } from './utils/constants/pageRoutes';

export const PROCESS_NAME = 'permits';

export const errorMsgMap: { [key: string]: string } = {
  'Incorrect emirates ID': 'errorMsg.incorrectemiratesid',
  'Incorrect Service Name': 'errorMsg.incorrectservicename',
  'Failed to send message': 'errorMsg.failedtosendmessage',
  'Incorrect DED license number': 'errorMsg.incorrectdedlicensenumber',
  'Incorrect pro mobile number': 'errorMsg.incorrectpromobilenumber',
  'Incorrect pro name': 'errorMsg.incorrectproname',
  'Incorrect email': 'errorMsg.incorrectemail',
  'Incorrect documents': 'errorMsg.incorrectdocuments',
  'Incorrect contact': 'errorMsg.incorrectcontact',
  'Incorrect Advertisement Details': 'errorMsg.incorrectadvertisementdetails',
  'Incorrect Advertisement Location': 'errorMsg.incorrectadvertisementlocation',
  'Incorrect Advertise Start Date': 'errorMsg.incorrectadvertisestartdate',
  'Incorrect Advertise End Date': 'errorMsg.incorrectadvertiseenddate',
  'Incorrect Customer Comment': 'errorMsg.incorrectcustomercomment',
  'Incorrect Study Specialization': 'errorMsg.incorrectstudyspecialization',
  'Incorrect Project Name': 'errorMsg.incorrectprojectname',
  'Incorrect Comments': 'errorMsg.incorrectcomments',
  'Incorrect Quantity': 'errorMsg.incorrectquantity',
  'Incorrect Plate Or Chassis Number': 'errorMsg.incorrectplateorchassisnumber',
  'Incorrect Seasonal Promotions': 'errorMsg.incorrectseasonalpromotions',
  'Incorrect Promotion Details': 'errorMsg.incorrectpromotiondetails',
  'Incorrect Promotion Location': 'errorMsg.incorrectpromotionlocation',
  'Incorrect Advertisements Quantity':
    'errorMsg.incorrectadvertisementsquantity',
  'Incorrect Plate Category': 'errorMsg.incorrectplatecategory',
  'Incorrect Advertisement Type': 'errorMsg.incorrectadvertisementtype',
  'Incorrect Note': 'errorMsg.incorrectnote',
  'Incorrect City': 'errorMsg.incorrectcity',
  'Incorrect Employee Comment': 'errorMsg.incorrectemployeecomment',
  'Incorrect Permit Type': 'errorMsg.incorrectpermittype',
  'Incorrect Airs': 'errorMsg.incorrectairs',
  'Incorrect Sign Boards': 'errorMsg.incorrectsignboards',
  'Please provide machines': 'errorMsg.pleaseprovidemachines',
  'Incorrect Umbrellas': 'errorMsg.incorrectumbrellas',
  'Incorrect Balloons': 'errorMsg.incorrectballoons',
  'Incorrect Sale Banners': 'errorMsg.incorrectsalebanners',
  'Incorrect Prizes': 'errorMsg.incorrectprizes',
  'Incorrect Propaganda Advertising Boards':
    'errorMsg.incorrectpropagandaadvertisingboards',
  'Incorrect Kiosks': 'errorMsg.incorrectkiosks',
  'Incorrect Vehicles': 'errorMsg.incorrectvehicles',
};

export const REQUIRES_SOP3 = {
  test: (props: IVariables) => {
    const { user } = props;
    if (user.Type === 'SOP3') return true;
    return false;
  },
  redirectTo: PATH_ACCOUNT_UPGRADE,
};

export const REQUIRES_CUSTOM_LOGIN = {
  test: (props: IVariables) => {
    if (props.loggedIn) return true;
    return false;
  },
  redirectTo: `${PATH_LOGIN}?redirectUrl=${window.location.href}`,
};
