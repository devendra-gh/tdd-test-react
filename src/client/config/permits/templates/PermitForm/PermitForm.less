.marginT20 {
  margin-top: 20px;
}

.marginT40 {
  margin-top: 40px;
}

.marginT30 {
  margin-top: 30px;
}

.marginT60 {
  margin-top: 60px;
}

.marginB0 {
  margin-bottom: 0 !important;
}

.marginB30 {
  margin-bottom: 30px;
}

.marginB20 {
  margin-bottom: 20px !important;
}

.ui-lib-text-area {
  .ui-lib-text-area__message {
    color: #fc532e;
  }
}

.ant-calendar-date::before {
  border: none !important;
}

.ui-lib-service-template__title {
  margin-top: 3rem !important;
}

.hide {
  display: none;
}
