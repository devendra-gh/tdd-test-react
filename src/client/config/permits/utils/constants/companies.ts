export const companies: { [key: string]: Record<string, any> } = {
  DED: {
    label: 'company.label.ded',
    id: 'DED',
    type: '104',
  },
  FZ: {
    label: 'company.label.fz',
    id: 'FZ',
    type: '104',
  },
  NL: {
    label: 'company.label.nl',
    id: 'NL',
    type: '100',
  },
  ADGE: {
    label: 'company.label.adge',
    id: 'ADGE',
    type: '13',
  },
  FC: {
    label: 'company.label.fc',
    id: 'FC',
    type: '12',
  },
};
