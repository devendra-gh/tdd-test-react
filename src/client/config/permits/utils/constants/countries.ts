export const countries = [
  {
    name: 'country.unitedarabemirates',
    code: 971,
  },
];
