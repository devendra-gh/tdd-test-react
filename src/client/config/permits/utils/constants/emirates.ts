export const emirates = [
  {
    id: '1',
    label: 'emirate.abuDhabi',
  },
  {
    id: '2',
    label: 'emirate.dubai',
  },
  {
    id: '3',
    label: 'emirate.sharjah',
  },
  {
    id: '4',
    label: 'emirate.ajman',
  },
  {
    id: '5',
    label: 'emirate.ummAlQuwain',
  },
  {
    id: '6',
    label: 'emirate.rasAlKhaima',
  },
  {
    id: '7',
    label: 'emirate.fujairah',
  },
];
