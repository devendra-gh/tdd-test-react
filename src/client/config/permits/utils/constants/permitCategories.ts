const PERMIT_CATEGORY_SIGNBOARD_AD = 'signboardAd';
const PERMIT_CATEGORY_ADDITIONAL_AD = 'additionalAdPermit';
const PERMIT_CATEGORY_MACHINE_AD = 'machineAd';
const PERMIT_CATEGORY_MOBILE_AD = 'mobileAd';
const PERMIT_CATEGORY_PROMOTIONAL_AD = 'promotionalAd';
export {
  PERMIT_CATEGORY_SIGNBOARD_AD,
  PERMIT_CATEGORY_ADDITIONAL_AD,
  PERMIT_CATEGORY_MACHINE_AD,
  PERMIT_CATEGORY_MOBILE_AD,
  PERMIT_CATEGORY_PROMOTIONAL_AD,
};
