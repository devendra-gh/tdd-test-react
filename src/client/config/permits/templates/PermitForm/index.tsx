import React, { useState } from 'react';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import Container from 'client/containers';
import Alert from '@tamm/ui-lib-v2-alert';
import Loading from 'client/templates/Loading';
import ServiceTemplate from '@tamm/ui-lib-v2-service-template';
import CompanyDetailsQuestion from './components/CompanyDetailsQuestion';
import PermitTypeForm from './components/PermitTypeForm';
import './PermitForm.less';

const margin = {
  margin: '20px',
};

const margin40 = {
  margin: '40px',
};

const PermitForm = (props: IVariables) => {
  const {
    permitInfo,
    companyDetails,
    companyType,
    permitCompanyTypes,
    handlePermitFormChange,
    handleCompanyTypeChange,
    handleRepresentativeTypeChange,
    handleCompanyDetailsChange,
    fetchLicenseDetails,
    serviceType,
    i18n,
  } = props;

  // const isWide = useMedia({ minWidth: '768px' });
  const [validate, setValidate] = useState(false);

  const onSubmit = () => {
    setValidate(true);
    const isValid = props.validation(props);
    if (!isValid) {
      return;
    }
    props.onSubmit(props);
    setValidate(false);
  };

  if (props.user && props.user.Type !== 'SOP3') {
    return (
      <Container locale={props.locale}>
        <div className={['ui-lib-margin-b_md'].join(' ')}>
          <Alert message={i18n('user_not_verified')} status="error" />
        </div>
        <div style={margin40} />
      </Container>
    );
  }

  return (
    <Container locale={props.locale}>
      {props.permitSubmitting && <Loading />}
      <div className={['PermitForm', 'ui-lib-margin-b_xl'].join(' ')}>
        {props.permitServerError ? (
          <div className="ui-lib-margin-b_md" id="error-message-div">
            <Alert message={i18n(props.permitServerError)} status="error" />
          </div>
        ) : (
          props.permitErrorMessage && (
            <div className="ui-lib-margin-b_md" id="error-message-div">
              <Alert
                message={props.permitErrorMessage.replace(/<[^>]+>/g, ' ')}
                status="warning"
              />
            </div>
          )
        )}
        <ServiceTemplate
          title={i18n('applicationdetails.templateTitle')}
          description={i18n('applicationdetails.templateDesc')}
        />
        <div style={margin} />
        <CompanyDetailsQuestion
          serviceType={serviceType}
          companyType={companyType}
          companyTypes={permitCompanyTypes}
          handleCompanyTypeChange={(value: any) =>
            handleCompanyTypeChange(props, value)
          }
          handleRepresentativeTypeChange={(value: any) =>
            handleRepresentativeTypeChange(props, value)
          }
          handleChange={(value: any) =>
            handleCompanyDetailsChange(props, value)
          }
          values={companyDetails}
          fetchLicenseDetails={fetchLicenseDetails}
          startShowingErrors={validate}
          i18n={i18n}
        />
        <div style={margin40} />
        <PermitTypeForm
          {...props}
          startShowingErrors={validate}
          handleChange={(value: any) => handlePermitFormChange(props, value)}
          values={permitInfo[serviceType]}
          onSubmit={onSubmit}
        />
        <div style={margin} />
      </div>
    </Container>
  );
};

export default withTemplateHooks(PermitForm);
