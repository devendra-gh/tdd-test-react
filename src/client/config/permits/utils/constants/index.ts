import { countries } from './countries';
import { legalForms } from './legalForms';
import { companies } from './companies';
import { emirates } from './emirates';
import { nationalities } from './nationalities';
import { permitType } from './permitType';
import { representativeType } from './representativeType';

export {
  companies,
  countries,
  emirates,
  legalForms,
  nationalities,
  permitType,
  representativeType,
};
