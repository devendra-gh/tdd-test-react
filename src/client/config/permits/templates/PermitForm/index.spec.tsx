import React, { FunctionComponent } from 'react';
import { shallow } from 'enzyme';
import { IVariables } from '@tamm/app-composer';
import PermitForm from './index';
import { PERMIT_CATEGORY_ADDITIONAL_AD } from '../../utils/constants/permitCategories';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: FunctionComponent) => component,
}));

const stateKey1 = 'permitDetails';
const permitType = 'airAdPermit';

const testPermitForm = {
  [stateKey1]: {
    label: 'label 1',
  },
};

const testPermits = {
  [permitType]: testPermitForm,
};

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Permit Form', () => {
  let props: any;

  beforeEach(() => {
    props = {
      form: {
        onChange: jest.fn(),
      },
      permitForm: testPermitForm,
      permitType,
      permits: testPermits,
      i18n: jest.fn(),
      onSubmit: jest.fn(),
      handleCategoryChange: jest.fn(),
      commonFieldConfig: {
        startDateLabel: 'Test label',
      },
      serviceType: PERMIT_CATEGORY_ADDITIONAL_AD,
      companyType: 'DED',
      handleServiceTypeChange: jest.fn(),
      handleCompanyDetailsChange: jest.fn(),
      handleCompanyTypeChange: jest.fn(),
      handlePermitFormChange: jest.fn(),
      getPermitCompanyTypes: jest.fn(),
      fetchLicenseDetails: jest.fn(),
      handleRepresentativeTypeChange: jest.fn(),
      actions: {
        validation: {
          reset: jest.fn(),
          update: jest.fn(),
        },
        companyDetails: {
          update: jest.fn(),
        },
      },
      validation: jest.fn(x => true),
      permitInfo: {
        airAdPermit: {
          companyTypes: {
            0: 'FZ',
          },
          permitDetails: {
            startDate: '',
            location: '',
            description: '',
          },
          applicantContact: {
            name: 'name',
            email: 'email',
            phone: 'mobile',
          },
          documents: {},
        },
      },
      companyDetails: {
        formValidated: false,
        licenseNo: '',
        licenseValid: false,
        location: '',
      },
      permitSubmitting: 'test',
      locale: 'testlocale',
      permitServerError: '',
    };
  });

  it('should render successfully', () => {
    const component = shallow(<PermitForm {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render successfully', () => {
    props.user = { type: 'SOP3' };
    const component = shallow(<PermitForm {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render successfully', () => {
    props.user = { type: 'SOP1' };
    const component = shallow(<PermitForm {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render successfully', () => {
    props.permitErrorMessage = '<a>';
    const component = shallow(<PermitForm {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render alert message', () => {
    props.permitServerError = 'error';
    const component = shallow(<PermitForm {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should call CompanyDetailsQuestion handle change', () => {
    const component = shallow(<PermitForm {...props} />);
    const CompanyDetails = component.find('CompanyDetailsQuestion');
    const allProps: any = CompanyDetails.props();
    const {
      handleChange,
      handleCompanyTypeChange,
      handleRepresentativeTypeChange,
    } = allProps;
    handleChange();
    expect(props.handleCompanyDetailsChange).toHaveBeenCalled();
    handleCompanyTypeChange();
    expect(props.handleCompanyTypeChange).toHaveBeenCalled();
    handleRepresentativeTypeChange();
    expect(props.handleRepresentativeTypeChange).toHaveBeenCalled();
  });

  it('should call PermitTypeForm handle change', () => {
    const component = shallow(<PermitForm {...props} />);
    const CompanyDetails = component.find('PermitTypeForm');
    const allProps: any = CompanyDetails.props();
    const { handleChange } = allProps;
    handleChange();
    expect(props.handlePermitFormChange).toHaveBeenCalled();
  });

  it('should call onSubmit validation true', () => {
    const component = shallow(<PermitForm {...props} />);
    const CompanyDetails = component.find('PermitTypeForm');
    const allProps: any = CompanyDetails.props();
    const { onSubmit } = allProps;
    onSubmit();
    expect(props.onSubmit).toBeCalled();
  });
  it('should call onSubmit validation false', () => {
    props.validation = (x: IVariables) => false;
    const component = shallow(<PermitForm {...props} />);
    const CompanyDetails = component.find('PermitTypeForm');
    const allProps: any = CompanyDetails.props();
    const { onSubmit } = allProps;
    onSubmit();
    expect(onSubmit).toBeInstanceOf(Function);
  });
});
