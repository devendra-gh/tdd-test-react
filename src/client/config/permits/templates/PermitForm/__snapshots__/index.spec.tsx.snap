// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permit Form should render alert message 1`] = `
<Container
  locale="testlocale"
>
  <Component />
  <div
    className="PermitForm ui-lib-margin-b_xl"
  >
    <div
      className="ui-lib-margin-b_md"
      id="error-message-div"
    >
      <Alert
        onClose={null}
        status="error"
      />
    </div>
    <ServiceTemplate
      description=""
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <div
      style={
        Object {
          "margin": "20px",
        }
      }
    />
    <CompanyDetailsQuestion
      companyType="DED"
      fetchLicenseDetails={[MockFunction]}
      handleChange={[Function]}
      handleCompanyTypeChange={[Function]}
      handleRepresentativeTypeChange={[Function]}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "error",
            ],
            Array [
              "applicationdetails.templateTitle",
            ],
            Array [
              "applicationdetails.templateDesc",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      serviceType="additionalAdPermit"
      startShowingErrors={false}
      values={
        Object {
          "formValidated": false,
          "licenseNo": "",
          "licenseValid": false,
          "location": "",
        }
      }
    />
    <div
      style={
        Object {
          "margin": "40px",
        }
      }
    />
    <PermitTypeForm
      actions={
        Object {
          "companyDetails": Object {
            "update": [MockFunction],
          },
          "validation": Object {
            "reset": [MockFunction],
            "update": [MockFunction],
          },
        }
      }
      commonFieldConfig={
        Object {
          "startDateLabel": "Test label",
        }
      }
      companyDetails={
        Object {
          "formValidated": false,
          "licenseNo": "",
          "licenseValid": false,
          "location": "",
        }
      }
      companyType="DED"
      fetchLicenseDetails={[MockFunction]}
      form={
        Object {
          "onChange": [MockFunction],
        }
      }
      getPermitCompanyTypes={[MockFunction]}
      handleCategoryChange={[MockFunction]}
      handleChange={[Function]}
      handleCompanyDetailsChange={[MockFunction]}
      handleCompanyTypeChange={[MockFunction]}
      handlePermitFormChange={[MockFunction]}
      handleRepresentativeTypeChange={[MockFunction]}
      handleServiceTypeChange={[MockFunction]}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "error",
            ],
            Array [
              "applicationdetails.templateTitle",
            ],
            Array [
              "applicationdetails.templateDesc",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      locale="testlocale"
      onSubmit={[Function]}
      permitForm={
        Object {
          "permitDetails": Object {
            "label": "label 1",
          },
        }
      }
      permitInfo={
        Object {
          "airAdPermit": Object {
            "applicantContact": Object {
              "email": "email",
              "name": "name",
              "phone": "mobile",
            },
            "companyTypes": Object {
              "0": "FZ",
            },
            "documents": Object {},
            "permitDetails": Object {
              "description": "",
              "location": "",
              "startDate": "",
            },
          },
        }
      }
      permitServerError="error"
      permitSubmitting="test"
      permitType="airAdPermit"
      permits={
        Object {
          "airAdPermit": Object {
            "permitDetails": Object {
              "label": "label 1",
            },
          },
        }
      }
      serviceType="additionalAdPermit"
      startShowingErrors={false}
      validation={[MockFunction]}
    />
    <div
      style={
        Object {
          "margin": "20px",
        }
      }
    />
  </div>
</Container>
`;

exports[`Permit Form should render successfully 1`] = `
<Container
  locale="testlocale"
>
  <Component />
  <div
    className="PermitForm ui-lib-margin-b_xl"
  >
    <ServiceTemplate
      description=""
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <div
      style={
        Object {
          "margin": "20px",
        }
      }
    />
    <CompanyDetailsQuestion
      companyType="DED"
      fetchLicenseDetails={[MockFunction]}
      handleChange={[Function]}
      handleCompanyTypeChange={[Function]}
      handleRepresentativeTypeChange={[Function]}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "applicationdetails.templateTitle",
            ],
            Array [
              "applicationdetails.templateDesc",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      serviceType="additionalAdPermit"
      startShowingErrors={false}
      values={
        Object {
          "formValidated": false,
          "licenseNo": "",
          "licenseValid": false,
          "location": "",
        }
      }
    />
    <div
      style={
        Object {
          "margin": "40px",
        }
      }
    />
    <PermitTypeForm
      actions={
        Object {
          "companyDetails": Object {
            "update": [MockFunction],
          },
          "validation": Object {
            "reset": [MockFunction],
            "update": [MockFunction],
          },
        }
      }
      commonFieldConfig={
        Object {
          "startDateLabel": "Test label",
        }
      }
      companyDetails={
        Object {
          "formValidated": false,
          "licenseNo": "",
          "licenseValid": false,
          "location": "",
        }
      }
      companyType="DED"
      fetchLicenseDetails={[MockFunction]}
      form={
        Object {
          "onChange": [MockFunction],
        }
      }
      getPermitCompanyTypes={[MockFunction]}
      handleCategoryChange={[MockFunction]}
      handleChange={[Function]}
      handleCompanyDetailsChange={[MockFunction]}
      handleCompanyTypeChange={[MockFunction]}
      handlePermitFormChange={[MockFunction]}
      handleRepresentativeTypeChange={[MockFunction]}
      handleServiceTypeChange={[MockFunction]}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "applicationdetails.templateTitle",
            ],
            Array [
              "applicationdetails.templateDesc",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      locale="testlocale"
      onSubmit={[Function]}
      permitForm={
        Object {
          "permitDetails": Object {
            "label": "label 1",
          },
        }
      }
      permitInfo={
        Object {
          "airAdPermit": Object {
            "applicantContact": Object {
              "email": "email",
              "name": "name",
              "phone": "mobile",
            },
            "companyTypes": Object {
              "0": "FZ",
            },
            "documents": Object {},
            "permitDetails": Object {
              "description": "",
              "location": "",
              "startDate": "",
            },
          },
        }
      }
      permitServerError=""
      permitSubmitting="test"
      permitType="airAdPermit"
      permits={
        Object {
          "airAdPermit": Object {
            "permitDetails": Object {
              "label": "label 1",
            },
          },
        }
      }
      serviceType="additionalAdPermit"
      startShowingErrors={false}
      validation={[MockFunction]}
    />
    <div
      style={
        Object {
          "margin": "20px",
        }
      }
    />
  </div>
</Container>
`;

exports[`Permit Form should render successfully 2`] = `
<Container
  locale="testlocale"
>
  <div
    className="ui-lib-margin-b_md"
  >
    <Alert
      onClose={null}
      status="error"
    />
  </div>
  <div
    style={
      Object {
        "margin": "40px",
      }
    }
  />
</Container>
`;

exports[`Permit Form should render successfully 3`] = `
<Container
  locale="testlocale"
>
  <div
    className="ui-lib-margin-b_md"
  >
    <Alert
      onClose={null}
      status="error"
    />
  </div>
  <div
    style={
      Object {
        "margin": "40px",
      }
    }
  />
</Container>
`;

exports[`Permit Form should render successfully 4`] = `
<Container
  locale="testlocale"
>
  <Component />
  <div
    className="PermitForm ui-lib-margin-b_xl"
  >
    <div
      className="ui-lib-margin-b_md"
      id="error-message-div"
    >
      <Alert
        message=" "
        onClose={null}
        status="warning"
      />
    </div>
    <ServiceTemplate
      description=""
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <div
      style={
        Object {
          "margin": "20px",
        }
      }
    />
    <CompanyDetailsQuestion
      companyType="DED"
      fetchLicenseDetails={[MockFunction]}
      handleChange={[Function]}
      handleCompanyTypeChange={[Function]}
      handleRepresentativeTypeChange={[Function]}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "applicationdetails.templateTitle",
            ],
            Array [
              "applicationdetails.templateDesc",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      serviceType="additionalAdPermit"
      startShowingErrors={false}
      values={
        Object {
          "formValidated": false,
          "licenseNo": "",
          "licenseValid": false,
          "location": "",
        }
      }
    />
    <div
      style={
        Object {
          "margin": "40px",
        }
      }
    />
    <PermitTypeForm
      actions={
        Object {
          "companyDetails": Object {
            "update": [MockFunction],
          },
          "validation": Object {
            "reset": [MockFunction],
            "update": [MockFunction],
          },
        }
      }
      commonFieldConfig={
        Object {
          "startDateLabel": "Test label",
        }
      }
      companyDetails={
        Object {
          "formValidated": false,
          "licenseNo": "",
          "licenseValid": false,
          "location": "",
        }
      }
      companyType="DED"
      fetchLicenseDetails={[MockFunction]}
      form={
        Object {
          "onChange": [MockFunction],
        }
      }
      getPermitCompanyTypes={[MockFunction]}
      handleCategoryChange={[MockFunction]}
      handleChange={[Function]}
      handleCompanyDetailsChange={[MockFunction]}
      handleCompanyTypeChange={[MockFunction]}
      handlePermitFormChange={[MockFunction]}
      handleRepresentativeTypeChange={[MockFunction]}
      handleServiceTypeChange={[MockFunction]}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "applicationdetails.templateTitle",
            ],
            Array [
              "applicationdetails.templateDesc",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      locale="testlocale"
      onSubmit={[Function]}
      permitErrorMessage="<a>"
      permitForm={
        Object {
          "permitDetails": Object {
            "label": "label 1",
          },
        }
      }
      permitInfo={
        Object {
          "airAdPermit": Object {
            "applicantContact": Object {
              "email": "email",
              "name": "name",
              "phone": "mobile",
            },
            "companyTypes": Object {
              "0": "FZ",
            },
            "documents": Object {},
            "permitDetails": Object {
              "description": "",
              "location": "",
              "startDate": "",
            },
          },
        }
      }
      permitServerError=""
      permitSubmitting="test"
      permitType="airAdPermit"
      permits={
        Object {
          "airAdPermit": Object {
            "permitDetails": Object {
              "label": "label 1",
            },
          },
        }
      }
      serviceType="additionalAdPermit"
      startShowingErrors={false}
      validation={[MockFunction]}
    />
    <div
      style={
        Object {
          "margin": "20px",
        }
      }
    />
  </div>
</Container>
`;
